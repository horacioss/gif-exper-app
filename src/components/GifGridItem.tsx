import { image } from "../types/image";

const GifGridItem: React.FC<image> = ({ url, title }) => (
  <div className="card animate__animated animated__fadein">
    <img src={url} alt={title} />
    <p className="title-image">{title}</p>
  </div>
);

export default GifGridItem;
