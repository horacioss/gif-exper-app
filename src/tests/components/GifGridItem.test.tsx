import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("tests over <GifGridItem />", () => {
  const img = {
    id: "43h9874d9gh53435g",
    title: "Any Title for an Image",
    url: "https://localhost/werr/4t.jpg",
  };

  let wrapper = shallow(<GifGridItem {...img} />);

  test("should render <GifGridItem />", () => {
    expect(wrapper).toMatchSnapshot();
  });
  /*
   * Testing properties
   */
  test("should paragraph to be equal at title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(img.title);
  });

  test("should image's alt & src to be equal to title url properties of img", () => {
    const image = wrapper.find("img");
    expect(image.prop("src")).toBe(img.url);
    expect(image.props().alt).toBe(img.title);
  });

  test("should contains animated__fadein in className", () => {
    const div = wrapper.find("div");
    expect(div.hasClass("animated__fadein")).toBe(true);
    const className = div.props().className;
    expect(className?.includes("animated__fadein")).toBe(true);
  });
});
