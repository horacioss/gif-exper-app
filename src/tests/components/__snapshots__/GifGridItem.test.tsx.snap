// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests over <GifGridItem /> should render <GifGridItem /> 1`] = `
<div
  className="card animated__animated animated__fadein"
>
  <img
    alt="Any Title for an Image"
    src="https://localhost/werr/4t.jpg"
  />
  <p
    className="title-image"
  >
    Any Title for an Image
  </p>
</div>
`;
